app/javascript/bundles/Groceries/components/Groceries.tsx

```diff
@@ -226,4 +229,15 @@ export default class Groceries extends React.Component<RouteComponentProps, Groc
 
     return itemList.map((item) => item.id == id ? { ...groceryItem, ...properties } : item)
   }
+
+  addNewItemToBasket(itemName: string, itemPrice: string): void {
+    const newItem: GroceryLineItem = { name: itemName, price: Number(itemPrice), quantity: 1, id: itemName }
+    const itemAlreadyExists: boolean = this.state.cartLineItems.find((item) => item.name === itemName) != undefined
```

Comment:
there's an array method `some` that will return true if some item in the array meets your criterion. Use that here instead.